src/unix/mmap.c

```diff
@@ -169,6 +168,33 @@ sysreturn mremap(void *old_address, u64 old_size, u64 new_size, int flags, void
     if (new_size <= old_size)
         return sysreturn_from_pointer(old_address);
 
+    /* verify we have a single vmap for the old address range */
+    vmap old_vm = (vmap)rangemap_lookup(p->vmaps, old_addr);
```

Comment:
I suppose there could be adjacent vmaps (e.g. an anonymous one and a fixed one immediately after) that are both affected by a remap. But we can track this with an issue and circle around...this should at least bail out sufficiently if we run into such a case.

Reply:
From `man 2 mremap:`

"EFAULT "Segmentation  fault."  Some  address  in the range old_address to old_address+old_size is an invalid virtual memory address for this process.  You can also get EFAULT even if there exist mappings that cover the whole address space requested, but those mappings are of different types."

So it seems like the adjacent maps issued of different types can safely fail. As to the other issues, yeah, if we see a need for them we can add them, shouldn't be too difficult.